lib/Cake/Test/Case/View/Helper/HtmlHelperTest.php

```diff
@@ -562,9 +562,15 @@ public function testBase64ImageTag()
 	{
 		$this->Html->request->webroot = '';
 
-		$result = $this->Html->image('cake.icon.png', ['base64' => true]);
 		//my apologies for the length of this...
-		$this->assertTags($result, array('img' => array('src' => 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABQAAAAUCAYAAACNiR0NAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA1FJREFUeNqsVF1IU2EYfraF5U951P0wnTidP/kzm5GZuygNDSWhQRnECMohhgnRnRcpKgQSqGhCEIhLJhkUGhGu0JwiXqSYQyddhG6EG7bQ/E1ccfq+D3fYULvyufgO53vf85zn/RXxPI+jxDF6iEQi4aK0tNRAHtdUKpUuOjpaJ5fLoVAomG1hYQFbW1twuVy2tbW1GXL1YnBwcMb/LRUnYgchpESxsbFtZWVl6vz8fIHkMFDyoaEhjIyMUPKHlJhFSw9C1mCxWPiDMDExwbe0tLDn5ubmPju9a2pq4glHgaCwtraWb25uPlRNTU0NUbwEj+cvpNIMpKVpQaNISkpidpoGk8k009fXl8NySPP0P1RVVaGr6xGKr4SSt0UsLzvR2/sBP5bFkCs0jDwiIoITikJzUVxcDK1WeyAhvc/IuEoUfoRSKYFKFYbUFBnCwmQQIRLfvtnhdrtZccR739jM5ka0traSvy8fSGo0GjE1KcGobQcu5x9IJCGQiEOoMkhjmDi7oJDg1+3b8SQXIeju7sTurgSJiUmIi1NBrVaD4yKxsrKKrKw8zM+P4uvX3xgetiM9XYnTaYlYX/dRjplAQrvF4jFUVuaguroQoaEq5rS46ITD4YDX62VOHMeRFhPDYEiAWMQR+y7mHN9JyG4mKpDQubq6i8eP36O8nEdJSRxIUyMmRhoU9tjYGN6988Fm8yL33Cmcz03FpYtSPO3sR3vHc1sQoUYTgdzcbFitn/HmzSdSOT30ej1rjfDwcObk8/mQnHwSUVFhGHg7T1IzhzPZqVhye51Bo0fjHx//SRxXUF9/DxsbxzE15cDAwAArEp0KP/R6OfLy4lFYcBY7OycwPe0iud8RCIXRM5mu8/HxMYzMaLwFnS6P5SsQVqsVHR0dNq1Woc6/kKnOyswgefSirf1VIxm9BsrlV0j6aNtcUXH5zvZ2JNrbn5F26EVRURE0Gg0CFwTB6JMn5kI6+0ql/YFcFlXgr3CQQuLAJSTIFuvq7nMyWQomJ+dZlWdnZwWFZG5p+HeJGnPAdlKTd6d/2yBwHxKj7ubNG196erp4j8e9bxH09/ezJXDYiAaFTLG323I2NrYML1++Fnai337YFAVCdNQbW4wjxj8BBgDSNqdSpFi+eQAAAABJRU5ErkJggg==', 'alt' => '')));
+		$reallyLongBase64Result = 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABQAAAAUCAYAAACNiR0NAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA1FJREFUeNqsVF1IU2EYfraF5U951P0wnTidP/kzm5GZuygNDSWhQRnECMohhgnRnRcpKgQSqGhCEIhLJhkUGhGu0JwiXqSYQyddhG6EG7bQ/E1ccfq+D3fYULvyufgO53vf85zn/RXxPI+jxDF6iEQi4aK0tNRAHtdUKpUuOjpaJ5fLoVAomG1hYQFbW1twuVy2tbW1GXL1YnBwcMb/LRUnYgchpESxsbFtZWVl6vz8fIHkMFDyoaEhjIyMUPKHlJhFSw9C1mCxWPiDMDExwbe0tLDn5ubmPju9a2pq4glHgaCwtraWb25uPlRNTU0NUbwEj+cvpNIMpKVpQaNISkpidpoGk8k009fXl8NySPP0P1RVVaGr6xGKr4SSt0UsLzvR2/sBP5bFkCs0jDwiIoITikJzUVxcDK1WeyAhvc/IuEoUfoRSKYFKFYbUFBnCwmQQIRLfvtnhdrtZccR739jM5ka0traSvy8fSGo0GjE1KcGobQcu5x9IJCGQiEOoMkhjmDi7oJDg1+3b8SQXIeju7sTurgSJiUmIi1NBrVaD4yKxsrKKrKw8zM+P4uvX3xgetiM9XYnTaYlYX/dRjplAQrvF4jFUVuaguroQoaEq5rS46ITD4YDX62VOHMeRFhPDYEiAWMQR+y7mHN9JyG4mKpDQubq6i8eP36O8nEdJSRxIUyMmRhoU9tjYGN6988Fm8yL33Cmcz03FpYtSPO3sR3vHc1sQoUYTgdzcbFitn/HmzSdSOT30ej1rjfDwcObk8/mQnHwSUVFhGHg7T1IzhzPZqVhye51Bo0fjHx//SRxXUF9/DxsbxzE15cDAwAArEp0KP/R6OfLy4lFYcBY7OycwPe0iud8RCIXRM5mu8/HxMYzMaLwFnS6P5SsQVqsVHR0dNq1Woc6/kKnOyswgefSirf1VIxm9BsrlV0j6aNtcUXH5zvZ2JNrbn5F26EVRURE0Gg0CFwTB6JMn5kI6+0ql/YFcFlXgr3CQQuLAJSTIFuvq7nMyWQomJ+dZlWdnZwWFZG5p+HeJGnPAdlKTd6d/2yBwHxKj7ubNG196erp4j8e9bxH09/ezJXDYiAaFTLG323I2NrYML1++Fnai337YFAVCdNQbW4wjxj8BBgDSNqdSpFi+eQAAAABJRU5ErkJggg==';
```

Comment:
I _could_ but then I'm nearly duplicating current code in a test. Thus if we ever change how something works natively, the test will have to change to match, rather than matching an expected output. I'm happy that if the icon is changed (which it's not been changed since 2009) then the test will break, as that's about the same as if you changed the code and the test broke. (Plus it's an easy fix to update the test)